lib/src/widgets/atomic/report_dialogue.dart

```diff
@@ -0,0 +1,131 @@
+import "package:btc_market/services.dart";
+import "package:btc_market/src/data/report.dart";
+import "package:flutter/material.dart";
+
+/// A report form.  
+/// Should be used as the content of an AlertDialogue.  
+class ReportDialogue extends StatefulWidget {
+  /// The type of item being reported
+  final ItemType itemType;
+
+  /// The ID of the item being reported
+  final String itemID;
+
+  /// Constructs a new report dialogue widget
+  const ReportDialogue({required this.itemType, required this.itemID, super.key});
+
+  @override
+  State<ReportDialogue> createState() => _ReportDialogueState();
+}
+
+/// A bunch of reasons that will appear on the dropdown menu,
+final reportReasons = [
+  "Other",
+  "Inappropriate/NSFW",
+];
+
+/// These ones depends on the type of item being reported
+final specificReportReasons = [
+  (
+    reason: "Troll", 
+    applicableTypes: [
+      ItemType.product, 
+      ItemType.sellerProfile, 
+      ItemType.userProfile,
+    ]
+  ),
+  (
+    reason: "Fraudulent activity", 
+    applicableTypes: [
+      ItemType.userProfile, 
+      ItemType.sellerProfile,
+    ]
+  ),
+  (
+    reason: "Disrespectful", 
+    applicableTypes: [
+      ItemType.userProfile, 
+      ItemType.sellerProfile,
+      ItemType.conversation,
+    ]
+  ),
+];
+
+class _ReportDialogueState extends State<ReportDialogue> {
+  final TextEditingController _textController = TextEditingController();
+  late String selectedReason;
+  late final List<String> choices;
+
+  @override
+  void initState() {
+    super.initState();
+    
+    choices = [
+      ...reportReasons, 
+      ...specificReportReasons
+        .where((record) => record.applicableTypes.contains(widget.itemType))
+        .map((record) => record.reason),
+    ];
```

Comment:
Very clean! If you wanted, you can also do: 

```dart
choices = [
  ...reportReasons,
  for (final (:reason, :applicableTypes) in specificReasons)
    if (applicableTypes.contains(widget.itemType) reason,  
];
```

(but again, I think we can keep all options for all item types based on the reasons you suggested